FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_Mutual_Auth/DemoTasks/MutualAuthHTTPExample.c

```diff
@@ -0,0 +1,461 @@
+ * connection.
+ *
+ * The example shown below uses HTTP APIs to first create a
+ * mutually-authenticated network connection with an HTTP server, and then send
+ * a POST request containing a simple message. This example is single threaded
+ * and uses statically allocated memory. It uses QoS1 for sending and receiving
+ * messages from the server.
+ *
+ * A mutually-authenticated TLS connection is used to connect to the HTTP server
+ * in this example. Define democonfigAWS_IOT_ENDPOINT, democonfigROOT_CA_PEM,
+ * democonfigCLIENT_PRIVATE_KEY_PEM, and democonfigCLIENT_CERTIFICATE_PEM in
+ * demo_config.h, to establish a mutually authenticated connection.
+ */
+
+/**
+ * @file MutualAuthHTTPExample.c
+ * @brief Demonstrates usage of the HTTP library.
+ */
+
+/* Standard includes. */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* Kernel includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+
+/* Demo Specific configs. */
+#include "demo_config.h"
+
+/* HTTP library includes. */
+#include "core_http_client.h"
+
+/* Exponential backoff retry include. */
+#include "exponential_backoff.h"
+
+/* Transport interface implementation include header for TLS. */
+#include "using_mbedtls.h"
+
+/* Common HTTP demo utilities. */
+#include "http_demo_utils.h"
+
+/*------------- Demo configurations -------------------------*/
+
+/* Check that the AWS IoT Core endpoint is defined. */
+#ifndef democonfigAWS_IOT_ENDPOINT
+    #error "Please define democonfigAWS_IOT_ENDPOINT in demo_config.h."
+#endif
+
+/* Check that the root CA certificate is defined. */
+#ifndef democonfigROOT_CA_PEM
+    #error "Please define democonfigROOT_CA_PEM in demo_config.h."
+#endif
+
+#ifndef democonfigCLIENT_CERTIFICATE_PEM
+    #error "Please define democonfigCLIENT_CERTIFICATE_PEM in demo_config.h."
+#endif
+
+#ifndef democonfigCLIENT_PRIVATE_KEY_PEM
+    #error "Please define democonfigCLIENT_PRIVATE_KEY_PEM in demo_config.h."
+#endif
+
+/* Check that a path for HTTP Method POST is defined. */
+#ifndef democonfigPOST_PATH
+    #error "Please define democonfigPOST_PATH in demo_config.h."
+#endif
+
+/* Check that a request body to send for the POST request is defined. */
+#ifndef democonfigREQUEST_BODY
+    #error "Please define a democonfigREQUEST_BODY in demo_config.h."
+#endif
+
+/* Check that a TLS port for AWS IoT Core is defined. */
+#ifndef democonfigAWS_HTTP_PORT
+    #define democonfigAWS_HTTP_PORT    ( 8443 )
+#endif
+
+/* Check that a transport timeout for transport send and receive is defined. */
+#ifndef democonfigTRANSPORT_SEND_RECV_TIMEOUT_MS
+    #define democonfigTRANSPORT_SEND_RECV_TIMEOUT_MS    ( 1000 )
+#endif
+
+/* Check that a size for the user buffer is defined. */
+#ifndef democonfigUSER_BUFFER_LENGTH
+    #define democonfigUSER_BUFFER_LENGTH    ( 2048 )
+#endif
+
+/**
+ * @brief The length of the AWS IoT Endpoint.
+ */
+#define httpexampleAWS_IOT_ENDPOINT_LENGTH    ( sizeof( democonfigAWS_IOT_ENDPOINT ) - 1 )
```

Comment:
Nit: It's slightly weird that we have lengths as an example. Maybe it would be best to just remove the prefix altogether.